test/cmocka/src/math/arithmetic/base_e_logarithm.c

```diff
@@ -0,0 +1,108 @@
+	21.3697795618183370, 21.3922524176471107, 21.4142313243436178, 21.4357375295432568,
+	21.4567909387206548, 21.4774102259037889, 21.4976129332024932, 21.5174155604805932,
+	21.5368336463203107, 21.5558818412742781, 21.5745739742703257, 21.5929231129229997,
+	21.6109416184107097, 21.6286411954956677, 21.6460329381935921, 21.6631273715394208,
+	21.6799344898427790, 21.6964637917813938, 21.7127243130127638, 21.7287246543415016,
+	21.7444730112924880, 21.7599771978118319, 21.7752446699265008, 21.7902825472754031,
+	21.8050976330453672, 21.8196964324517815, 21.8340851698895619, 21.8482698048676056,
+	21.8622560468288185, 21.8760493689479993, 21.8896550209909755, 21.9030780413106569,
+	21.9163232680485471, 21.9293953496040821, 21.9422987544284780, 21.9550377801946830,
+	21.9676165623906030, 21.9800390823784895, 21.9923091749598925, 22.0044305354820757,
+	22.0164067265188734, 22.0282411841561903, 22.0399372239099236, 22.0514980465667030,
+	22.0629267423782807, 22.0742262976204415, 22.0853995982070579, 22.0964494343823858,
+	22.1073785049035614, 22.1181894209970018, 22.1288847101032040, 22.1394668194234150,
+	22.1499381192805984, 22.1603009063062437, 22.1705574064637361, 22.1807097776854185};
+
+/* testvector in Q32.0 */
+static const uint32_t uv[100] = {
+	1ULL,	 43383509ULL,	86767017ULL,   130150525ULL, 173534033ULL, 216917541ULL,
+	260301049ULL, 303684557ULL,  347068065ULL,  390451573ULL, 433835081ULL, 477218589ULL,
+	520602097ULL, 563985605ULL,  607369113ULL,  650752621ULL, 694136129ULL, 737519638ULL,
+	780903146ULL, 824286654ULL,  867670162ULL,  911053670ULL, 954437178ULL, 997820686ULL,
+	1041204194ULL, 1084587702ULL, 1127971210ULL, 1171354718ULL, 1214738226ULL, 1258121734ULL,
+	1301505242ULL, 1344888750ULL, 1388272258ULL, 1431655766ULL, 1475039274ULL, 1518422782ULL,
+	1561806290ULL, 1605189798ULL, 1648573306ULL, 1691956814ULL, 1735340322ULL, 1778723830ULL,
+	1822107338ULL, 1865490846ULL, 1908874354ULL, 1952257862ULL, 1995641370ULL, 2039024878ULL,
+	2082408386ULL, 2125791894ULL, 2169175403ULL, 2212558911ULL, 2255942419ULL, 2299325927ULL,
+	2342709435ULL, 2386092943ULL, 2429476451ULL, 2472859959ULL, 2516243467ULL, 2559626975ULL,
+	2603010483ULL, 2646393991ULL, 2689777499ULL, 2733161007ULL, 2776544515ULL, 2819928023ULL,
+	2863311531ULL, 2906695039ULL, 2950078547ULL, 2993462055ULL, 3036845563ULL, 3080229071ULL,
+	3123612579ULL, 3166996087ULL, 3210379595ULL, 3253763103ULL, 3297146611ULL, 3340530119ULL,
+	3383913627ULL, 3427297135ULL, 3470680643ULL, 3514064151ULL, 3557447659ULL, 3600831168ULL,
+	3644214676ULL, 3687598184ULL, 3730981692ULL, 3774365200ULL, 3817748708ULL, 3861132216ULL,
+	3904515724ULL, 3947899232ULL, 3991282740ULL, 4034666248ULL, 4078049756ULL, 4121433264ULL,
+	4164816772ULL, 4208200280ULL, 4251583788ULL, 4294967295ULL};
+
+/*constant for base e Q2.30 */
+static const int32_t base_e = BASE_E_NATURAL_LOGARITM;
+
+static void test_math_arithmetic_base_e_log_fixed(void **state)
```

Comment:
@cujomalainey btw, do we have fuzzing on the cmocka logic. Might be worth a thought as the a lot of the maths logic uses table with array lookups (which depend on the runtime inputs).

Reply:
We do not, but we can definitely add endpoints to fuzz the math functions directly quite easily, we wouldnt even need cmocka

Reply:
Do I need to take any action here ?

Reply:
> we wouldnt even need cmocka

Would the fuzzer test the correctness of logarithm functions too?

Reply:
no, it would simply look for inputs that rely on undefined behaviours or push pointers out of bounds, or crash the program